### PR TITLE
Refactor run/primitives page

### PR DESCRIPTION
**Pull request by abbycross**
Supersedes #84 

Closes https://github.com/Qiskit/documentation/issues/254

### CODE REVIEW

**Comment by abbycross** [comment]
I replaced the content with material from the blog post @javabster wrote.

Then I incorporated the "difference between open-source/runtime primitives" box since it doesn't exist elsewhere in the docs (but maybe there's a better place for it??)

Then I added a line about estimating and sampling being fundamental to quantum computation, which is one of the requests in #254. I imagine it needs some supporting evidence though! 😆

**Comment by ElePT** [comment]
Hello! I have pushed a new version of the page with quite a lot of changes. It is considerably longer too, which I am not sure is the intention for this page, but I thought it could be nice to have a deeper contextualization of why we wanted to introduce primitives in the first place. Please take a look and let me know if the general direction seems better. If it's worse we can always revert to the previous changes.

File: docs/run/primitives.mdx [hunk 1]
```diff
@@ -6,68 +6,88 @@ description: Introduction to primitives in Qiskit and Qiskit Runtime, and an exp
 
 # Introduction to primitives
 
-In computing, a primitive is the smallest processing instruction for a given abstraction level. Put simply, they are the simplest building blocks from which a user can create something useful - and since primitives abstract away implementation details, users can accomplish tasks without needing to worry about low-level specifics.  The computational model proposed by primitives moves quantum programming one step closer to where classical programming is today, where the focus is less on the hardware details and more on the results you are trying to achieve. 
+Computing systems are built upon multiples layers of **abstraction**. Abstractions generalize certain
```
Comment (ElePT) [inline docs/run/primitives.mdx:9]:
```suggestion
Computing systems are built upon multiple layers of **abstraction**. Abstractions generalize certain
```

File: docs/run/primitives.mdx [hunk 2]
```diff
@@ -6,68 +6,88 @@ description: Introduction to primitives in Qiskit and Qiskit Runtime, and an exp
 
 # Introduction to primitives
 
-In computing, a primitive is the smallest processing instruction for a given abstraction level. Put simply, they are the simplest building blocks from which a user can create something useful - and since primitives abstract away implementation details, users can accomplish tasks without needing to worry about low-level specifics.  The computational model proposed by primitives moves quantum programming one step closer to where classical programming is today, where the focus is less on the hardware details and more on the results you are trying to achieve. 
+Computing systems are built upon multiple layers of **abstraction**. Abstractions generalize certain
+details of *how* the technology works, to allow us to focus our attention on the details that are relevant for a
+specific task. The level of abstraction we choose to work with will depend on how we want to interact with the device.
+The closer we get to the hardware, the lower-level we need the abstractions to be (for example, we could
+want to manipulate electrical signals), and vice-versa, the more complex the task we want to perform,
```
Comment (abbycross) [inline docs/run/primitives.mdx:13]:
```suggestion
want to manipulate electrical signals), and vice versa, the more complex the task we want to perform,
```

**Comment by abbycross** [comment]
Thanks so much @ElePT, this reads very well!  I started to make small style-based tweaks but didn't want to clutter up your PR with comments, so I will wait until I'm sure you're not working on the file at the same time, and I'll go through and make little changes. All I want to address are:

lower-case P for primitives (I think it is only capitalized twice), and
change the voice from "we" to "you" where possible

I'll leave it to @javabster to comment on the substance of the content, but from a writing perspective it looks good.

**Comment by ElePT** [comment]
Thanks [abbycross](https://github.com/abbycross), I've applied the feedback.

File: docs/run/primitives.mdx [hunk 3]
```diff
@@ -6,68 +6,88 @@ description: Introduction to primitives in Qiskit and Qiskit Runtime, and an exp
...
+If a provider does not support primitives natively, you can use these classes to “wrap” any backend into a primitive. Users can write primitive-style code for providers that don’t yet have a primitives-based interface. These classes can be used just like the regular Sampler and Estimator, except they should be initialized with an additional backend argument for selecting which backend to run on.
 
-* The primitives in ``qiskit_ibm_runtime`` provide access to cloud simulators and real hardware through the Qiskit Runtime service. They include exclusive features such as built-in circuit optimization and error mitigation support.
+The Qiskit Runtime primitives provide a more sophisticated implementation (such as with error mitigation) as a cloud-based service.
 
 ## Next steps
 
 <Admonition type="tip" title="Recommendations">
```
Comment (javabster) [inline docs/run/primitives.mdx:90]: Can you add a link to the "get started with primitives" page in the next steps section please?

**Comment by javabster** [comment]
This is looking great from my perspective @ElePT! 

@blakejohnson or @jaygambetta may have some more specific feedback (you can view the preview here: https://qiskit-docs-preview-pr-293.1799mxdls7qz.us-south.codeengine.appdomain.cloud/run/primitives)

File: docs/run/primitives.mdx [hunk 4]
```diff
@@ -6,68 +6,88 @@ description: Introduction to primitives in Qiskit and Qiskit Runtime, and an exp
...
-    - Review detailed [primitives examples.](../run/primitives-examples)
+    - Review detailed [primitives examples.](primitives-examples)
```
Comment (abbycross) [inline docs/run/primitives.mdx:91]:
```suggestion
    - Read [Get started with primitives](primitives-get-started) to implement primitives in your work.
    - Review detailed [primitives examples.](primitives-examples)
```

File: docs/run/primitives.mdx (hunk 3 above)
Comment (abbycross) [reply docs/run/primitives.mdx:90]: I set up a suggested addition for that link here https://github.com/Qiskit/documentation/pull/293/files#r1391373158

File: docs/run/primitives.mdx [hunk 5]
```diff
@@ -6,68 +6,88 @@ description: Introduction to primitives in Qiskit and Qiskit Runtime, and an exp
...
+variances. Different Estimator implementations support various configuration options. The circuits
+can be parametrized, as long as the parameter values are also provided as input to the primitive.
 
-## Should I use Estimator or Sampler? 
-     
-When choosing which primitive to use, you first need to understand whether the algorithm uses a quasi-probability distribution sampled from a quantum state (a list of quasi-probabilities), or an expectation value of a certain observable with respect to a quantum state (a real number).
+## Sampler
 
-A probability distribution is often of interest in optimization problems that return a classical bit string, encoding a certain solution to a problem at hand. In these cases, you might be interested in finding a bit string that corresponds to a ket value with the largest probability of being measured from a quantum state, for example.
+The Sampler primitive samples results from a series of measurements performed on states prepared by quantum circuits.
```
Comment (blakejohnson) [inline docs/run/primitives.mdx:61]: The "series of measurements" is unclear. I suggest instead:
```suggestion
The Sampler primitive samples results from the classical output registers resulting from execution of quantum circuits.
```

**Review by blakejohnson — APPROVED** [review]
Just the one comment. Otherwise, LGTM.

**Comment by javabster** [comment]
According to the link checker the removal of the "import-primitives" span caused a link to break in the verify section, so I've included in this PR a fix to that page as well. I opted to just remove the whole line in the verify page talking about imports as we no longer explicitly talk about primitives imports here

**Pull request merged**